File: go-client-sdk/pkg/models/operations/getintegrationscredentialsplatformkey.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetIntegrationsCredentialsPlatformKeyPathParams struct {
	PlatformKey string `pathParam:"style=simple,explode=false,name=platformKey"`
}

type GetIntegrationsCredentialsPlatformKeySecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetIntegrationsCredentialsPlatformKeyRequest struct {
	PathParams GetIntegrationsCredentialsPlatformKeyPathParams
	Security   GetIntegrationsCredentialsPlatformKeySecurity
}

type GetIntegrationsCredentialsPlatformKeyResponse struct {
	ContentType                                                   string
	StatusCode                                                    int
	GetIntegrationsCredentialsPlatformKey200ApplicationJSONObject map[string]interface{}
}

File: typescript-client-sdk/src/sdk/models/operations/getintegrationscredentialsplatformkey.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class GetIntegrationsCredentialsPlatformKeyPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=platformKey" })
  platformKey: string;
}

export class GetIntegrationsCredentialsPlatformKeySecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetIntegrationsCredentialsPlatformKeyRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetIntegrationsCredentialsPlatformKeyPathParams;

  @SpeakeasyMetadata()
  security: GetIntegrationsCredentialsPlatformKeySecurity;
}

export class GetIntegrationsCredentialsPlatformKeyResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  getIntegrationsCredentialsPlatformKey200ApplicationJSONObject?: Record<string, any>;
}